src/profile-logic/profile-data.js

```diff
@@ -879,38 +880,41 @@ export function filterThreadToSearchString(
 }
 
 /**
- * This function takes both a SamplesTable and can be used on CounterSamplesTable.
+ * This function takes a SamplesTable and can also be used on CounterSamplesTable.
+ * It takes a time-based range as parameter and returns a sample-based range.
+ * Just like rangeStart is inclusive and rangeEnd is exclusive, sampleStart is
+ * inclusive and sampleEnd is exclusive.
  */
-function _getSampleIndexRangeForSelection(
+export function getSampleIndexRangeForSelection(
   samples: SamplesTable | CounterSamplesTable,
   rangeStart: number,
   rangeEnd: number
-): [IndexIntoSamplesTable, IndexIntoSamplesTable] {
+): StartEndSampleRange {
   // TODO: This should really use bisect. samples.time is sorted.
-  const firstSample = samples.time.findIndex(t => t >= rangeStart);
-  if (firstSample === -1) {
-    return [samples.length, samples.length];
+  const sampleStart = samples.time.findIndex(t => t >= rangeStart);
+  if (sampleStart === -1) {
+    return { sampleStart: samples.length, sampleEnd: samples.length };
   }
-  const afterLastSample = samples.time
-    .slice(firstSample)
+  const sampleEnd = samples.time
+    .slice(sampleStart)
     .findIndex(t => t >= rangeEnd);
-  if (afterLastSample === -1) {
-    return [firstSample, samples.length];
+  if (sampleEnd === -1) {
+    return { sampleStart, sampleEnd: samples.length };
   }
-  return [firstSample, firstSample + afterLastSample];
+  return { sampleStart, sampleEnd: sampleEnd + sampleStart };
 }
```

Comment:
I didn't change the implementation, just renamed variables and changed the return type to an object instead of a tuple.
I initially wanted to use bisect like the TODO says, but then decided that it should be in a separate PR because I wasn't confident about the edge cases and how our bisection library works. I think this needs a separate PR with tests for edge cases.

Reply:
https://github.com/firefox-devtools/profiler/issues/2139

I filed this before I saw your comment. That's an old TODO.